### PR TITLE
1019 enable staff to update fosterer info

### PR DESCRIPTION
**Pull request by Meri-MG**
# 🔗 Issue
#1019 

# ✍️ Description

- Added update and edit action to the `FosterersController`
- Added `update_fosterers` ability to `FostererInvitationPolicy` for update? and edit? actions(I figured nobody should be seeing the edit form if they don't have permission to perform the update action?)
- Removed `Joined At` column from the fosterers table
- Added email, and phone columns to the table and left action column for the Edit link
- Moved form errors into the partial

P.S. @kasugaijin I'm unsure why we are advised to use turbo frames here?

# 📷 Screenshots/Demos
Screenshot 2024-10-10 at 13 03 43
Screenshot 2024-10-09 at 14 17 14

dave

### CODE REVIEW

**Comment by kasugaijin** [comment]
@Meri-MG thanks for the PR - it might take me a couple days before I get to review it - busy times. Hopefully someone else can also drop a review.

**Comment by kasugaijin** [comment]
@Meri-MG no need to use turbo frames! That was optional. The redirects work fine. We could use a frame to handle it all in the same page without redirects, but that is not necessary. 

Could you please update the edit buttons to match those used elsewhere? See the staff FAQ page for example.
image


I will do a full review, soon!

File: app/views/organizations/staff/fosterers/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<%= render DashboardPageComponent.new(crumb: :edit_fosterer, crumb_options: [@fosterer]) do |c| %>
+  <% c.with_header_title { t(".edit_fosterer", fosterer: @fosterer.full_name) } %>
+  <% c.with_body do %>
+    <%= render "form", { fosterer: @fosterer } %>
+  <% end %>
+<% end %>
```
Comment (kasugaijin) [inline app/views/organizations/staff/fosterers/edit.html.erb:6]: new line :)

File: test/controllers/organizations/staff/fosterers_controller_test.rb [hunk 2]
```diff
@@ -39,4 +40,49 @@ class Organizations::Staff::FosterersControllerTest < ActionDispatch::Integratio
       end
     end
   end
+
+  context "#edit" do
```
Comment (kasugaijin) [inline test/controllers/organizations/staff/fosterers_controller_test.rb:44]: There's separate test files for asserting authorization see `test/policies/organizations/fosterer_invitation_policy_test.rb`

**Review by kasugaijin — COMMENTED** [review]
Looks good @Meri-MG just a couple small comments

File: app/views/partials/_form_errors.html.erb
Comment (kasugaijin) [inline app/views/partials/_form_errors.html.erb:1]: i like this idea...but maybe let's keep the error rendering on the form. Then we can do a sweep of all forms and figure out the best place for this to live. Thoughts?

**Comment by Meri-MG** [comment]
hello, @kasugaijin , had some busy days and addressed the feedback just now. thank you 🙏

File: app/views/organizations/staff/fosterers/edit.html.erb [hunk 3]
```diff
@@ -3,4 +3,4 @@
   <% c.with_body do %>
     <%= render "form", { fosterer: @fosterer } %>
   <% end %>
-<% end %>
\ No newline at end of file
+<% end %>
```
Comment (kasugaijin) [inline app/views/organizations/staff/fosterers/edit.html.erb:6]: New line :)

File: test/controllers/organizations/staff/fosterers_controller_test.rb (hunk 2 above)
Comment (kasugaijin) [inline test/controllers/organizations/staff/fosterers_controller_test.rb:44]: I think we can remove these tests testing authorization now you have added the policy tests?

Comment (Meri-MG) [reply test/controllers/organizations/staff/fosterers_controller_test.rb:44]: oh, okay, I wanted to recreate what was written for the index action in this tests.

File: app/views/organizations/staff/fosterers/edit.html.erb (hunk 3 above)
Comment (Meri-MG) [reply app/views/organizations/staff/fosterers/edit.html.erb:6]: weird thing is happening for me on these lines, unless I add two empty lines, the new line doesn't show up 🤔 

Screenshot 2024-10-18 at 11 34 28

Comment (kasugaijin) [reply app/views/organizations/staff/fosterers/edit.html.erb:6]: Weird! It's just showing one line in github so we will roll with this

**Review by kasugaijin — APPROVED** [review]
Awesome job! Thank you @Meri-MG

**Pull request merged**